### PR TITLE
Add filters in the url

### PR DESCRIPTION
**Pull request by pmareke**
EX-3511

### CODE REVIEW

File: packages/x-components/src/x-modules/search/store/actions/set-params-from-url.action.ts [hunk 1]
```diff
@@ -24,4 +26,8 @@ export const setParamsFromUrl: SearchXStoreModule['actions']['setParamsFromUrl']
   if (newPage) {
     commit('setPage', newPage);
   }
+
+  if (newFilters) {
+    commit('setSelectedFilters', createRawFilters(newFilters));
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/search/store/actions/set-params-from-url.action.ts:31]: Is this necessary? Doesn't the facet's module communicate the selected filters in time for the first request?

File: packages/x-components/src/x-modules/facets/store/actions/set-filters-from-url.action.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+/**
+ * Default implementation for the {@link FacetsActions.setUrlParamsFromTheUrl}.
+ *
+ * @param context - The {@link https://vuex.vuejs.org/guide/actions.html | context} of the actions,
+ * provided by Vuex.
+ * @param urlParams - The url params from the url.
+ *
+ * @public
+ */
+export const setUrlParamsFromTheUrl: FacetsXStoreModule['actions']['setUrlParamsFromTheUrl'] = (
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/facets/store/actions/set-filters-from-url.action.ts:14]: Why is this file called `set-filters-from-url.action`, but the exported method `setUrlParamsFromTheUrl`?. Both names should match the action name

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/facets/store/actions/set-filters-from-url.action.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { createRawFilters, Dictionary } from '../../../../utils';
+import { UrlParamValue } from '../../../url';
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/facets/store/actions/set-filters-from-url.action.ts:2]: Avoid importing from barrels. It is very easy to create cyclical dependencies that can result in errors at runtime. In this case you are even importing from a barrel that should be exporting this action, which is worse.

File: packages/x-components/src/x-modules/facets/store/actions/set-filters-from-url.action.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ */
+export const setUrlParamsFromTheUrl: FacetsXStoreModule['actions']['setUrlParamsFromTheUrl'] = (
+  { commit },
+  { query, filters }: Dictionary<UrlParamValue>
+) => {
+  const newQuery = query as string;
+  const newFilters = filters as string[];
+
+  if (newQuery) {
+    commit('setQuery', newQuery);
```
Comment (tajespasarela) [inline packages/x-components/src/x-modules/facets/store/actions/set-filters-from-url.action.ts:22]: Why is necessary the `query` in the `Facets` module?

I only see it in an emitter of a `FacetsQueryChanged` Event which is never used.

I would remove all of it: the query, the emitter and the Event

File: packages/x-components/src/x-modules/facets/store/types.ts [hunk 5]
```diff
@@ -99,7 +101,13 @@ export interface FacetsMutations {
  *
  * @public
  */
-export interface FacetsActions {}
+export interface FacetsActions {
+  /**
+   * Receives a list of params from the url, builds the entities and
+   * set them in the store.
+   */
+  setUrlParamsFromTheUrl(urlParams: Dictionary<UrlParamValue>): void;
```
[the remark this answers is not in the document]
Comment (tajespasarela) [reply packages/x-components/src/x-modules/facets/store/types.ts:109]: I would keep the `set` verb, to keep consistency in all mutations/actions we have:
`setFiltersFromUrl`

As I said up, I would remove the query and then this action only set the filters.

File: packages/x-components/src/x-modules/search/store/actions/set-params-from-url.action.ts (hunk 1 above)
Comment (tajespasarela) [reply packages/x-components/src/x-modules/search/store/actions/set-params-from-url.action.ts:31]: And if not, we have to fix it in another way. 

We can not have the facets module and the search module doing the same.

File: packages/x-components/src/x-modules/url/store/module.ts [hunk 6]
```diff
@@ -44,6 +44,12 @@ export const urlXStoreModule: UrlXStoreModule = {
     setRelatedTags(state, relatedTags) {
       state.params.relatedTags = relatedTags;
     },
+    setFilters(state, newFilters) {
+      state.params.filters = newFilters;
+    },
+    resetFilters(state) {
```
[the remark this answers is not in the document]
Comment (tajespasarela) [reply packages/x-components/src/x-modules/url/store/module.ts:50]: right, remove this mutation

**Review by tajespasarela — CHANGES_REQUESTED** [review]

File: packages/x-components/src/components/scroll/base-scroll.vue [hunk 7]
```diff
@@ -58,7 +58,7 @@
      *
      * @internal
      */
-    @XOn(['UserAcceptedAQuery', 'SortChanged'])
+    @XOn(['UserAcceptedAQuery', 'SortChanged', 'UserPickedARelatedTag'])
```
[the remark this answers is not in the document]
Comment (pmareke) [reply packages/x-components/src/components/scroll/base-scroll.vue:61]: I think we should change algo the `UserPickedARelatedTag ` right?

File: packages/x-components/src/x-modules/facets/service/__tests__/facets-service.spec.ts [hunk 8]
```diff
@@ -296,7 +297,12 @@ describe('testing facets service', () => {
           getStoreEditableNumberRangeFilter(priceUpTo10Filter)
         ])
       ).toBe(false);
-      expect(getSelectedFilters()).toEqual([]);
+      expect(getSelectedFilters()).toEqual([
```
[the remark this answers is not in the document]
Comment (pmareke) [reply packages/x-components/src/x-modules/facets/service/__tests__/facets-service.spec.ts:300]: The problem is that the test is broken and the `[]` is not the expected response, that's why I did the change.

[the remark this answers is not in the document]
Comment (pmareke) [reply packages/x-components/src/x-modules/facets/service/__tests__/facets-service.spec.ts:300]: I rollback the entire spec file as @javieri-empathy suggested just changing `updateFacets` is ok.

File: packages/x-components/src/x-modules/url/wiring.ts [hunk 9]
```diff
@@ -37,6 +44,22 @@ export const setRelatedTagsWire = mapWire(
...
+ * Sets the {@link @empathyco/x-types#Filter | filters }.
+ *
+ * @public
+ */
+export const setFiltersWire = mapWire(wireCommit('setFilters'), (filters: Filter[]) =>
+  filters.map(f => f.id)
+);
+
+/**
+ * Resets the {@link RelatedTag | related tags }.
```
[the remark this answers is not in the document]
Comment (pmareke) [reply packages/x-components/src/x-modules/url/wiring.ts:57]: Removed it.

File: packages/x-components/src/x-modules/facets/store/actions/set-filters-from-url.action.ts (hunk 2 above)
Comment (pmareke) [reply packages/x-components/src/x-modules/facets/store/actions/set-filters-from-url.action.ts:14]: Fixed.

**Review by tiborux — CHANGES_REQUESTED** [review]

**Comment by tiborux** [comment]
Remove the FacetsQueryChanged from the event.types.ts in the facets module

File: packages/x-components/src/x-modules/facets/wiring.ts [hunk 10]
```diff
@@ -74,11 +74,11 @@ const clearAllFiltersWire = wireFacetsServiceWithoutPayload('clearFilters');
 const selectFilterWire = wireFacetsService('select');
 
 /**
- * Sets the facets state `query`.
+ * Saves the params from the url.
  *
  * @public
  */
-const setFacetsQuery = facetsWireCommit('setQuery');
+export const setFiltersFromUrl = facetsWireDispatch('setFiltersFromUrl');
```
Comment (tiborux) [inline packages/x-components/src/x-modules/facets/wiring.ts:81]: is this exported for any reason?

**Review by tiborux — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/facets/wiring.ts (hunk 10 above)
Comment (pmareke) [reply packages/x-components/src/x-modules/facets/wiring.ts:81]: My fault.

**Review by tiborux — APPROVED** [review]

**Review by guillei10 — APPROVED** [review]

**Pull request merged**